Source/Boxed.AspNetCore.TagHelpers/SubresourceIntegrityTagHelper.cs

```diff
@@ -245,5 +250,37 @@ private Task SetCachedSriAsync(string url, string value)
             var bytes = this.ReadAllBytes(filePath);
             return GetSpaceDelimetedSri(bytes, hashAlgorithms);
         }
+
+        private string GetEncodedStringValue(object attributeValue)
+        {
+            if (attributeValue is string stringValue)
+            {
+                var encodedStringValue = this.htmlEncoder.Encode(stringValue);
+                return encodedStringValue;
+            }
+            else
+            {
+                if (attributeValue is IHtmlContent htmlContent)
+                {
+                    if (htmlContent is HtmlString htmlString)
+                    {
+                        // No need for a StringWriter in this case.
+                        stringValue = htmlString.ToString();
+                    }
+                    else
+                    {
+                        using (var writer = new StringWriter())
+                        {
+                            htmlContent.WriteTo(writer, this.htmlEncoder);

```

Comment:
Should this be async?

Reply:
I'm not sure how you would do that, `IHtmlContent` doesnt have an async write method. And `StringWriter` does have an `WriteAsync()` but I dont see a way to specify the encoder. I might be missing something though.

Reply:
So IHtmlContent has no async methods, ok. Do we need to call `stringWriter.FlushAsync()`?

Reply:
On a StringWriter I do not think Flush actually does anything.